src/ng1/directives/marqueeWizard/marqueeWizard.controller.js

```diff
@@ -169,6 +191,10 @@ export default function MarqueeWizardCtrl($scope) {
     if (vm.stepIndex > 0 && vm.buttonOptions.showPrevious === true) vm.showPrevious = true;
     if (vm.stepIndex === vm.steps.length - 1 && vm.buttonOptions.showFinish === true) vm.showFinish = true;
     else if (vm.buttonOptions.showNext) vm.showNext = true;
+
+    //check for hidden steps
+    vm.showFinish = vm.steps.slice(vm.stepIndex).filter(step => !step.hidden).length === 0;
+    vm.showNext = !vm.showFinish;
```

Comment:
Finish button doesn't show on last step when i skip steps:

![image](https://user-images.githubusercontent.com/20795331/27188501-443f0f60-51e6-11e7-9d05-141342e8f6f1.png)

You may need to change this code a bit:

```javascript
vm.showFinish = vm.steps.slice(vm.stepIndex).filter(step => !step.hidden).length === 0
```

This might work, but i haven't tested it. It will be something along these lines

```javascript
vm.showFinish = vm.steps.slice(vm.stepIndex).filter(step => step.hidden !== true).length === 0
```